coverage.py

```diff
@@ -100,18 +102,31 @@ def log(self):
         return self._log        
 
     def run(self):
-        self.log.info("%d items need coverage.", (
-            self.items_that_need_coverage().count())
-        )
+        self.run_once_and_update_timestamp()
+
+    def run_once_and_update_timestamp(self):
+
+        # First cover items that have never had a coverage attempt
+        # before.
         offset = 0
         while offset is not None:
-            offset = self.run_once_and_update_timestamp(offset)
+            offset = self.run_once(
+                offset, covered_statuses=BaseCoverageRecord.ALL_STATUSES
+            )
 
-    def run_once_and_update_timestamp(self, offset):
-        offset = self.run_once(offset)
+        # Next, cover items that failed with a transient failure
+        # on a previous attempt.
+        offset = 0
+        while offset is not None:
+            offset = self.run_once(
+                offset, covered_statuses=[
+                    BaseCoverageRecord.SUCCESS, 
+                    BaseCoverageRecord.PERSISTENT_FAILURE
+                ]
+            )
+        
         Timestamp.stamp(self._db, self.service_name)
         self._db.commit()
-        return offset
```

Comment:
The concept of a singular 'offset' no longer makes sense since we are actually running two different sets of batches. Fortunately I don't think this was being used by run().